tests/test_job.py

```diff
@@ -435,6 +432,9 @@ def test_interface_multiple_changes(self):
             assert job.sp == job2.sp
             assert job.id == job2.id
 
+    @pytest.mark.filterwarnings("ignore:Use of int as key is deprecated")
+    @pytest.mark.filterwarnings("ignore:Use of NoneType as key is deprecated")
+    @pytest.mark.filterwarnings("ignore:Use of bool as key is deprecated")
```

Comment:
Please separate this and many other unrelated changes relating to filterwarnings marks into a separate PR.